### DISCUSSION TITLE
Filter for multi-hop archiving

### DISCUSSION THREAD

**Discussion by ga-it** [ArchiveBox/ArchiveBox#950] [Ideas]
It would be amazing if ArchiveBox could be set to crawl and archive against a pattern.

For example, for content behind a paywall, it would be very useful to pull all links within a subdirectory, e.g. https://foo.bar/articles/*

A further filtering using plain text or regex on page content within the target domain(s) and directories, e.g. all pages containing the word "leadership"

Regards

Marc

**Comment by pirate** [comment]
You can combine `URL_WHITELIST` and `URL_BLACKLIST` and `--depth=1` to achieve this.

```bash
URL_WHITELIST="^https://foo.bar/articles/.*$"
URL_BLACKLIST="..."
archivebox config --get URL_WHITELIST   # check that it's set correctly

archivebox add --depth=1 'https://foo.bar/articles'
```

https://github.com/ArchiveBox/ArchiveBox/wiki/Configuration#url_blacklist
https://github.com/ArchiveBox/ArchiveBox/wiki/Configuration#url_whitelist
